### ISSUE TITLE
`PROGRAM_FILE` not set when compiled through `juliac.jl`

### ISSUE THREAD

**Issue by NHDaly** [JuliaLang/PackageCompiler.jl#54]
`Base.PROGRAM_FILE` is empty when compiled through `juliac.jl`. (The global `ARGS` is empty, which is where `PROGRAM_FILE` is set from.)

`PROGRAM_FILE` is set [here](https://github.com/JuliaLang/julia/blob/v0.6.2/base/client.jl#L301), by popping the first value off of `ARGS`, iff `ARGS` is not empty (checked [here](https://github.com/JuliaLang/julia/blob/33fdd3/base/client.jl#L271)).

Currently, the default `program.c` skips setting the global `ARGS`, opting instead to [pass an Array to `julia_main`](https://github.com/JuliaLang/PackageCompiler.jl/blob/b8f0d0a/examples/program.c#L32-L40). This means that there is no way to access `PROGRAM_FILE` from compiled code.

**Comment by NHDaly** [comment]
Here is an example:

```julia
# examples/args.jl

foo() = println("ARGS: $ARGS")  # Sometimes code accesses ARGS directly, as a global

# Create a temporary .html file, and open it to share the greetings.
Base.@ccallable function julia_main(args::Vector{String})::Cint
    println("@__DIR__: $(@__DIR__)")
    println("@__FILE__: $(@__FILE__)")
    println("PROGRAM_FILE: $(PROGRAM_FILE)")
    foo()  # access global ARGS
    return 0
end

julia_main(["a b c"])
```

```bash
$ julia examples/args.jl d e f
@__DIR__: /Users/daly/src/build-jl-app-bundle/examples
@__FILE__: /Users/daly/src/build-jl-app-bundle/examples/args.jl
PROGRAM_FILE: examples/args.jl
ARGS: String["d", "e", "f"]  # NOTE: these are the command line args, not the inputs to `julia_main()`.
```

```bash
$ julia ~/.julia/v0.6/PackageCompiler/juliac.jl -v -a -e examples/args.jl && ./builddir/args d e f
...
@__DIR__: /Users/daly/src/build-jl-app-bundle/examples    # Note these are not a viable replacement (compile-time macros)
@__FILE__: /Users/daly/src/build-jl-app-bundle/examples/args.jl
PROGRAM_FILE:
ARGS: String[]
```

**Comment by NHDaly** [comment]
@lucatrv: I'm interested in your thoughts here as well. Do you have any idea how we could set global `ARGS` and/or `PROGRAM_FILE`?

Does the method I used in my [custom program.c, above](https://github.com/NHDaly/ApplicationBuilder.jl/commit/95098a4b82ffbed04a2ff748f33f319c536ec950#diff-0e48c24b6c52db1e06db58eb5c62d917R48) seem like a hack, or should I send a PR to add it to this repo's `examples/program.c`?

I hope your travel was nice!! 😄

**Comment by lucatrv** [comment]
I think we can add the fix for `PROGRAM_FILE = argv[0]`, since we already pass all the following `argv` elements to `julia_main`.

Yes my travel was very nice, thanks!!

**Comment by NHDaly** [comment]
Oh wowwwww i've always wanted to go to japan!!! That's excellent! :)

(opened PR #57)

**Comment by NHDaly** [comment]
Fixed in #57
